contracts/strategies/morpho/MetaMorphoStrategy.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "morpho-urd/src/interfaces/IUniversalRewardsDistributor.sol";
+import "../../interfaces/IERC20Detailed.sol";
+import "../../interfaces/morpho/IMetamorphoSnippets.sol";
+import "../ERC4626Strategy.sol";
+
+contract MetaMorphoStrategy is ERC4626Strategy {
+  using SafeERC20Upgradeable for IERC20Detailed;
+
+  //// @notice MORPHO governance token
+  address internal constant MORPHO = 0x9994E35Db50125E0DF82e4c2dde62496CE330999;
+
+  /// @notice reward token address (e.g. MORPHO, COMP, ...)
+  address[] public rewardTokens;
+
+  /// @notice MORPHO transferability
+  bool public morphoTransferable;
+
+  /// @notice Metamorpho snippets contract, used for apr
+  address public mmSnippets;
```

Comment:
Unlike the libraries we implemented (which could/should be imported and used by integrators if needed), snippets are contracts that only highlights use cases and shows how to interact with MetaMorpho (and Morpho Blue).

But snippets contracts should not be deployed (we won't deploy any snippet contract on any chain, and strongly advise anyone else not to). If you want to use a snippet function (in this case `supplyAPYVault`) I would advise you to implement the function directly in this contract (you can copy/paste from the snippet contract).

Reply:
Just FYI @Jean-Grimal https://github.com/Idle-Labs/idle-tranches/pull/83#discussion_r1431773939

Reply:
@Jean-Grimal I was talking with Tom a couple of weeks ago and in theory a single instance of the contract was supposed to be deployed once for all integrators so to avoid having to copy paste all the code for all integrators multiple times. Having eg the `supplyAPYVault(address vault)` with the `vault` parameter was an idea that I suggested to Tom to allow a single deployment which can then serve multiple vaults and reduce a bit the integrators work

Reply:
Okay I didn't know about that, thank you for this information